gb-ppu/src/drawing/state.rs

```diff
@@ -0,0 +1,121 @@
+use super::Mode;
+use crate::registers::LcdReg;
+use std::cell::RefMut;
+
+pub struct State {
+    mode: Mode,
+    line: u8,
+    step: u16,
+    pixel_drawn: u8,
+}
+
+impl State {
+    const LINE_COUNT: u8 = 154;
+    const LAST_LINE: u8 = Self::LINE_COUNT - 1;
+    const VBLANK_START: u8 = 144;
+
+    const PIXEL_DRAWING_START: u16 = 80;
+    const LAST_OAM_FETCH_STEP: u16 = Self::PIXEL_DRAWING_START - 1;
+    const HBLANK_MIN_START: u16 = 252;
+    const HBLANK_MAX_START: u16 = 369;
+    const STEP_COUNT: u16 = 456;
+    const LAST_STEP: u16 = Self::STEP_COUNT - 1;
+
+    pub fn new() -> Self {
```

Comment:
if a `new()` takes no arguments then it should be a `Default::default()`?

Reply:
no necessarily, but I can add a `Default` implementation
[constructor](https://rust-unofficial.github.io/patterns/idioms/ctor.html)
[default trait](https://rust-unofficial.github.io/patterns/idioms/default.html)